code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/DynamicSchemaDaoImpl.java

```diff
@@ -67,4 +150,47 @@ public int deleteSchemas(long serviceId) {
                 ).fetch().intoArray(INSTANCE.AGENT_ID));
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public void createRoles(DynamicSchema dynamicSchema) {
```

Comment:
For this and delete, you are executing more than a single query. you need to wrap in a transaction